### PR TITLE
Fix TransitDecryptResponse interfaces

### PR DESCRIPTION
**Pull request by LukasFritzeDev**
Fix TransitDecryptResponse interfaces to match the vault response for empty strings.

### CODE REVIEW

File: src/engines/transit_types.ts [hunk 1]
```diff
@@ -4,6 +4,10 @@ export type ITransitBatchPlaintext = Array<{
     plaintext: string;
     context?: string;
 }>;
+export type ITransitRawBatchPlaintext = Array<{
+    plaintext?: string;
```
Comment (martin-helmich) [inline src/engines/transit_types.ts:8]: Maybe add a link to the respective Vault issue, to explain why this curious typing is necessary?

Comment (martin-helmich) [reply src/engines/transit_types.ts:8]: Oh, nevermind. Just noticed that there's a reference to that issue elsewhere.

File: src/engines/transit.ts [hunk 2]
```diff
@@ -226,6 +228,33 @@ export class TransitVaultClient extends AbstractVaultClient {
      */
     public async decryptText(key: string, ciphertext: string): Promise<string> {
         validateKeyName(key);
-        return this.decrypt(key, { ciphertext }).then((res) => Buffer.from(res.data.plaintext, "base64").toString());
+        return this.decrypt(key, { ciphertext }).then((res) => {
+            if (res.data.plaintext === undefined) {
+                return "";
+            }
+            return Buffer.from(res.data.plaintext, "base64").toString();
+        });
```
Comment (martin-helmich) [inline src/engines/transit.ts:236]: Since this method is `async` anyway, can we to this without chaining promises, just using `await`?

```suggestion
        const res = await this.decrypt(key, { ciphertext });
        if (res.data.plaintext === undefined) {
            return "";
        }
        return Buffer.from(res.data.plaintext, "base64").toString();
```

File: tests/engines/transit.spec.ts [hunk 3]
```diff
@@ -95,7 +95,10 @@ describe("Transit Vault Client", () => {
                 .then((res) => res.data.batch_results);
             const res = await client
                 .decrypt("test", {
-                    batch_input: result,
+                    batch_input: result.map((r) => ({
+                        ciphertext: r.ciphertext,
+                        context: r.context,
+                    })),
```
Comment (martin-helmich) [inline tests/engines/transit.spec.ts:101]: Maybe add another test case that explicitly tests for empty plaintexts.

**Review by martin-helmich — CHANGES_REQUESTED** [review]

File: tests/engines/transit.spec.ts (hunk 3 above)
Comment (LukasFritzeDev) [reply tests/engines/transit.spec.ts:104]: Good idea :) done

File: src/engines/transit_types.ts (hunk 1 above)
Comment (LukasFritzeDev) [reply src/engines/transit_types.ts:8]: 👍

**Review by martin-helmich — APPROVED** [review]
